packages/design-system/src/components/Tooltip/Tooltip.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import classNames from 'classnames';
+import { autoPlacement, useFloating, offset, autoUpdate, shift } from '@floating-ui/react-dom';
+import useId from '../utilities/useId';
+import { Button } from '../Button';
+import { CloseButton } from '../CloseButton';
+
+// export interface BaseTooltipProps {
+//   block?: boolean;
+//   children: React.ReactNode;
+//   label: string;
+//   id?: string;
+//   // placement?: Placement;
+// }
+
+// // Similarly to the Button component, we want to expand the props type definition to
+// // permit pass-through props for the most commonly used underlying components.
+// type OtherProps = Omit<
+//   // All other props that could be passed to buttons
+//   React.ComponentPropsWithRef<'button'>,
+//   // Omit any properties that we're defining on our own `BaseTooltipProps`
+//   keyof BaseTooltipProps
+// >;
+
+// export type TooltipProps = BaseTooltipProps & OtherProps;
+
+/**
+ * Tooltips provide additional information upon hover, focus or click.
+ * For information about how and when to use this component,
+ * [refer to its full documentation page](https://design.cms.gov/components/tooltip/).
+ *
+ * When using the `<TooltipIcon>` as the only child of `<Tooltip>`, be sure to
+ * provide an `aria-label` on the `<Tooltip>` to ensure an accessible name for
+ * the trigger.
+ */
+export const Tooltip = (props) => {
+  const tooltipId = useId('tooltip-trigger--', props.id);
+
+  const { refs, floatingStyles } = useFloating({
+    middleware: [
+      autoPlacement({ allowedPlacements: ['top', 'bottom'] }),
+      offset(props.offset),
+      shift(),
+    ],
+    whileElementsMounted: autoUpdate,
```

Comment:
There's a little flicker where the tooltip gets repositioned after it first renders. This is a pretty complicated library, so I imagine it won't be a trivial task to fix that.